sourcery: init at 1.15.0

TomaSajt opened the pull request:
###### Description of changes

Closes: https://github.com/NixOS/nixpkgs/issues/243774
Closes: https://github.com/NixOS/nixpkgs/issues/244143

This PR adds the Sourcery.ai Python package and the VSCode extension.
Although the repo says it's MIT licensed, the Pypi pacakge says it's proprietary (which it actually is).

For the Python package, there are only platform specific wheels available on Pypi.
I don't know which platforms are supported actually, so I'll just go with x86_64 for linux and darwin.

The VSCode extension has all platforms' sources in it, so hopefully that will work as expected.

https://github.com/NixOS/nixpkgs/issues/243774 said that the package name was `python-sourcery`, but that's another, unrelated package

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

jfvillablanca commented:
Result of `nixpkgs-review pr 244908` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python310Packages.sourcery</li>
    <li>python310Packages.sourcery.dist</li>
    <li>python311Packages.sourcery</li>
    <li>python311Packages.sourcery.dist</li>
    <li>vscode-extensions.sourcery.sourcery</li>
  </ul>
</details>

MaxwellDupre reviewed (approved):
LGTM, except more likely to get merged if only one commit.

TomaSajt commented:
Well, it's been a while since I took a better look at this PR.
I rebased to a more recent commit.
I updated the version to `1.15.0`
I moved the package to `pkgs/by-name` while using `buildPythonApplication` instead of treating this as a normal python package (because it is not).
I also combined the two commits into one.

TomaSajt commented:
@ofborg build sourcery

TomaSajt commented:
@ofborg build sourcery

MaxwellDupre reviewed (approved):
Looks good.

superherointj commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -23,6 +23,7 @@
 , typst-preview
 , autoPatchelfHook
 , zlib
+, libxcrypt-legacy
```
Recently has been documented the review criteria for `vscode-extensions`, you might find it useful:

> https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vscode/extensions/README.md#conventions-for-adding-new-extensions

Consider:
> Move extension to a discrete directory whenever the extension needs extra parameters/packages (at top of the file) or other files (such as patches, update script, components). Global index file parameters/packages should be utilities shared by many extensions. Extension specific parameters/packages should not be in the global index page.

superherointj commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -3217,6 +3218,32 @@ let
+          name = "sourcery";
+          publisher = "sourcery";
+          version = "1.15.0";
+          sha256 = "sha256-aRUg/MZfMj9ZbuGy5CZGZ1LGHC5KMAVYWQAbe0scHYQ=";
```
```suggestion
          hash = "sha256-aRUg/MZfMj9ZbuGy5CZGZ1LGHC5KMAVYWQAbe0scHYQ=";
```

superherointj commented on pkgs/applications/editors/vscode/extensions/default.nix:
```diff
@@ -3217,6 +3218,32 @@ let
+          stdenv.cc.cc.lib
+          libxcrypt-legacy
+        ];
+        meta = {
```
changelog, downloadPage, homepage is missing?

superherointj commented:
Please split `vscode-extension` package into it's own commit.

superherointj commented:
For formatting of **new** Nix files (not for files that existed already, since that would cause lots of changes), I currently recommend using `nixfmt-rfc-style` (available in `nixos-unstable`).

superherointj commented on pkgs/by-name/so/sourcery/package.nix:
```diff
@@ -0,0 +1,43 @@
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "sourcery";
+  version = "1.15.0";
```
```suggestion
  version = "1.16.0";
```

superherointj commented:
> I also combined the two commits into one.

It should be a commit per package.

superherointj commented:
The commit message of Python package is missing the `release` reference. Like:

> https://github.com/sourcery-ai/sourcery/releases/tag/v1.16.0

superherointj commented:
Also, remember to rebase.

TomaSajt committed:
sourcery: init at 1.16.0

https://github.com/sourcery-ai/sourcery/releases/tag/v1.16.0

TomaSajt committed:
vscode-extensions.sourcery.sourcery: init at 1.16.0

TomaSajt commented:
I'm not sure if `downloadPage` should be set to what I set it to currently, because you'd probably download it using `pip` or from the `vscode` extension store.

TomaSajt commented:
Another thing I noticed: this is not the only package with the `sourcery` name.
https://repology.org/projects/?search=sourcery
https://github.com/krzysztofzablocki/Sourcery
It has more stars on GitHub
Should I rename the base-package to something else? (e.g. `sourcery-ai`)

superherointj commented:
> I'm not sure if `downloadPage` should be set to what I set it to currently, because you'd probably download it using `pip` or from the `vscode` extension store.

- `downloadPage` for the vscode-extension would be link to the vscode marketplace.
- `downloadPage` for the python package would be a link to the pypi page.
Homepage is usually where the source code is kept. (GitHub, GitLab, etc)

superherointj commented:
The python package should match `pypi` package name.

superherointj commented:
> I'm not sure if `downloadPage` should be set to what I set it to currently, because you'd probably download it using `pip` or from the `vscode` extension store.

Having this meta information is important to ease upkeep. For example, a maintainer can just click the link to see current situation of package in pypi/vscode marketplace, etc. There are other uses for it. Like the bot will add this information to PR opened. And it can provide meaningful information to anyone looking into the package. In general, we should add it. I don't see a reason to not add it.

superherointj commented:
Do you still want to call application `sourcery-ai`? You can do that if you wish.

TomaSajt commented:
> Do you still want to call application `sourcery-ai`? You can do that if you wish.

Well, I don't know what's better. I guess I'll keep it as is.
